taichi/analysis/data_source_analysis.cpp

```diff
@@ -0,0 +1,42 @@
+#include "taichi/ir/ir.h"
+#include "taichi/ir/analysis.h"
+#include "taichi/ir/statements.h"
+
+TLANG_NAMESPACE_BEGIN
+
+namespace irpass::analysis {
+
+Stmt *get_data_source(Stmt *store_stmt) {
```

Comment:
```suggestion
// Returns the *data* being stored to the destination, in store_stmt.
Stmt *get_data_source(Stmt *store_stmt) {
```

Or maybe rename it to `get_store_data`?

Reply:
Yeah `get_store_data` and `get_store_destination` are better names.